### PR TITLE
Fixing namespace for good w/o reexport + fix `x` instead of `tt`

### LINKED ISSUE #152: Address `export_as_txt` refactor
> Note this will require a change to teal.widgets https://github.com/insightsengineering/teal.widgets/blob/main/R/table_with_settings.R#L217 unless you re-export it through rtables - please co-ordinate with @donyunardi so that we don't say release teal.widgets and then it not work with a new release of rtables

_Originally posted by @nikolas-burkoff in https://github.com/insightsengineering/NEST-roadmap/issues/84#issuecomment-1488534728_
            
### Summary
The `export_as_txt()` function is undergoing refactoring and relocation to the formatters R package, which will have an impact on the `table_with_settings_ui/srv` functions within teal.widgets. This particular function is heavily utilized across multiple teal modules.

### Acceptance Criteria
* Assess if we can still use the `export_as_txt` function from formatters
* If yes, update the code base to use `export_as_txt` from formatters
* If not, propose solution to address this issue.

### PR DESCRIPTION
**Pull request by Melkiades**
Fixes #152 

Please feel free to add in to this if needed

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -34,7 +34,7 @@ Suggests:
     magrittr,
     png,
     rmarkdown,
-    rtables (>= 0.5.1),
+    rtables (>= 0.6.0.9004),
```
Comment (Melkiades) [inline DESCRIPTION:37]: why is this in the suggests? @donyunardi

Comment (donyunardi) [reply DESCRIPTION:37]: I am not too sure. 

Judging from this line:
https://github.com/insightsengineering/teal.widgets/blob/5c1bc697a2c422e6041cbb85bf80614316b83adb/R/table_with_settings.R#L78-L80

and several other lines that is using rtables function in `basic_table_args.R`, it sounds plausible to move rtables from Suggest to Imports.

I'll open an issue so we can discuss and not holding up this PR.

Comment (m7pr) [reply DESCRIPTION:37]: I've also seen this approach in other packages. Look at this one - shiny is in Suggests as it is only used in one of of 30 functions. shiny has at least 20 dependencies, so by adding shiny to Imports you would actually be adding 20 packages with all their dependencies, just to support one function. It's a tradeoff that is quite common in R world
https://github.com/pbiecek/archivist/blob/master/DESCRIPTION#L42
https://github.com/pbiecek/archivist/blob/master/R/shinySearchInLocalRepo.R#L57

**Comment by donyunardi** [comment]
> or we should add
> 
> ```
>   if (!requireNamespace("rtables", quietly = TRUE)) {
>     stop("package rtables is required, please install")
>   }
> ```
> 
> at the beginning of `basic_table_args` so there is the same information and consistency with `table_with_settings`. Then if someone does not uses tables/rtables then he/she does not need to install rtables to use teal.widgets

That makes perfect sense!
Adding your comment to the issue that I created.

**Pull request merged**